Yokogawa WT310E ranging failures at under 0.2 A

I'm investigating a very weird issue with the updated r1.1 workflow. It started with testing workflows on an Edge system. In the Offline scenario, the system power is over 24 Watts, and everything works as expected. In the Single Scenario, however, the system power is under 20 Watts, and I'm unable to take the measurements. Here's a typical failure log on the director side:

```
08-01-2021 10:56:59.225: WARNING: Uncertainty unknown for the last measurement sample! channel 0
ptd-server 2021-08-01 11:56:59,416 [INFO] Sending to ptd: 'Stop'
08-01-2021 10:56:59.417: Response to client sent: Stopping untimed measurement
ptd-server 2021-08-01 11:56:59,417 [INFO] Reply from ptd: 'Stopping untimed measurement'
ptd-server 2021-08-01 11:56:59,437 [ERROR] Got an exception
Traceback (most recent call last):
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 577, in handle_connection
    reply = self._handle_cmd(cmd, p)
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 650, in _handle_cmd                                     
    return unbool[self.session.stop(Mode.RANGING)]
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 861, in stop                                                                                
    self._maxVolts, self._maxAmps = max_volts_amps(
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 187, in max_volts_amps           
    raise MaxVoltsAmpsNegativeValuesError(f"Could not find values for {mark!r}")
ptd_client_server.lib.server.MaxVoltsAmpsNegativeValuesError: Could not find values for '2021-08-01_11-45-44_ranging'                                                                                      
ptd-server 2021-08-01 11:56:59,437 [INFO] Sending reply to client 'Error: exception'
ptd-server 2021-08-01 11:56:59,452 [INFO] Connection closed         
ptd-server 2021-08-01 11:56:59,452 [WARNING] Client connection closed unexpectedly
ptd-server 2021-08-01 11:56:59,452 [INFO] Sending to ptd: 'Stop'
08-01-2021 10:56:59.923: Completed test
08-01-2021 10:56:59.923: WARNING: No valid watts samples for this measurement!      
08-01-2021 10:56:59.924: Response to client sent: Error: no measurement to stop
ptd-server 2021-08-01 11:56:59,924 [INFO] Reply from ptd: 'Error: no measurement to stop'
ptd-server 2021-08-01 11:56:59,924 [INFO] Sending to ptd: 'SR,V,300.000000'
08-01-2021 10:56:59.924: Volt range set to 300.000000V
```

Here's how the PTDaemon log looks like on the director side:

```
me,08-01-2021 10:56:42.225,Watts,-22.100000,Volts,248.180000,Amps,-3.168000,PF,0.456200,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:43.226,Watts,-21.900000,Volts,248.130000,Amps,-3.168000,PF,0.454700,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:44.225,Watts,-21.300000,Volts,248.150000,Amps,-3.163000,PF,0.452500,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:45.225,Watts,-16.800000,Volts,248.130000,Amps,-3.130000,PF,0.427200,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:46.226,Watts,-16.000000,Volts,248.060000,Amps,-3.124000,PF,0.421700,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:47.225,Watts,-16.000000,Volts,248.140000,Amps,-3.125000,PF,0.422000,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:48.225,Watts,-15.700000,Volts,248.140000,Amps,-3.122000,PF,0.419100,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:49.226,Watts,-15.600000,Volts,248.170000,Amps,-3.121000,PF,0.417600,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:50.225,Watts,-15.500000,Volts,248.170000,Amps,-3.121000,PF,0.417300,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:51.225,Watts,-15.100000,Volts,248.090000,Amps,-3.118000,PF,0.413500,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:52.225,Watts,-15.200000,Volts,248.070000,Amps,-3.119000,PF,0.413400,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:53.225,Watts,-15.200000,Volts,248.110000,Amps,-3.119000,PF,0.413100,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:54.226,Watts,-15.200000,Volts,248.180000,Amps,-3.118000,PF,0.414100,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:55.225,Watts,-15.200000,Volts,248.220000,Amps,-3.119000,PF,0.414600,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:56.225,Watts,-15.200000,Volts,248.210000,Amps,-3.118000,PF,0.414400,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:57.226,Watts,-15.200000,Volts,248.230000,Amps,-3.118000,PF,0.415100,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:58.225,Watts,-15.200000,Volts,248.110000,Amps,-3.118000,PF,0.415300,Mark,2021-08-01_11-45-44_ranging
Time,08-01-2021 10:56:59.225,Watts,-15.200000,Volts,248.100000,Amps,-3.118000,PF,0.415200,Mark,2021-08-01_11-45-44_ranging
```
The power figures are negative but otherwise sensible. The Amps figures are negative and non-sensible.

@s-idgunji As I explained above, for systems that draw less than 0.2 A, the ranging log is full of negative and non-sensible values. The workflow doesn't even allow one to proceed to the testing phase failing with:

```
ptd-server 2021-08-01 11:56:59,437 [ERROR] Got an exception
Traceback (most recent call last):
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 577, in handle_connection
    reply = self._handle_cmd(cmd, p)
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 650, in _handle_cmd                                     
    return unbool[self.session.stop(Mode.RANGING)]
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 861, in stop                                                                                
    self._maxVolts, self._maxAmps = max_volts_amps(
  File "/home/anton/projects/mlperf/power-dev-r1.1/ptd_client_server/../ptd_client_server/lib/server.py", line 187, in max_volts_amps           
    raise MaxVoltsAmpsNegativeValuesError(f"Could not find values for {mark!r}")
ptd_client_server.lib.server.MaxVoltsAmpsNegativeValuesError: Could not find values for '2021-08-01_11-45-44_ranging'                                                                                      
ptd-server 2021-08-01 11:56:59,437 [INFO] Sending reply to client 'Error: exception'
```
If we don't fix it, this will affect hundreds of submissions, including customer-critical ones.

--- a/compliance/sources_checksums.json
+++ b/compliance/sources_checksums.json
@@ -32,5 +32,22 @@
         "tests/unit/__init__.py": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
         "tests/unit/test_server.py": "99ae15aef722f2000ee6ed1ae1523637bf1ae42b",
         "tests/unit/test_source_hashes.py": "00468a2907583c593e6574a1f6b404e4651c221a"
+    },
+    {
+        "client.py": "33ca4f26368777ac06e01f9567b714a4b8063886",
+        "server.py": "c3f90f2f7eeb4db30727556d0c815ebc89b3d28b",
+        "__init__.py": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
+        "lib/summary.py": "aa92f0a3f975eecd44d3c0cd0236342ccc9f941d",
+        "lib/client.py": "4c2b78fb4849a7e5b584ef792d82aaed20b17f57",
+        "lib/server.py": "59c9fb92908260d3e9f81a895c7aa009742275e4",
+        "lib/source_hashes.py": "60a2e02193209e8d392803326208d5466342da18",
+        "lib/__init__.py": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
+        "lib/common.py": "624d0c0acc7c39aaff3674f0b99d6a09da53d1dc",
+        "lib/time_sync.py": "3210db56eb0ff0df57bf4293dc4d4b03fffd46f1",
+        "lib/external/ntplib.py": "4da8f970656505a40483206ef2b5d3dd5e81711d",
+        "lib/external/__init__.py": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
+        "tests/unit/test_server.py": "99ae15aef722f2000ee6ed1ae1523637bf1ae42b",
+        "tests/unit/test_source_hashes.py": "00468a2907583c593e6574a1f6b404e4651c221a",
+        "tests/unit/__init__.py": "da39a3ee5e6b4b0d3255bfef95601890afd80709"
     }
 ]

--- a/ptd_client_server/lib/server.py
+++ b/ptd_client_server/lib/server.py
@@ -63,15 +63,6 @@ MULTICHANNEL_DEVICES = [48, 59, 61, 77]
 # https://github.com/mlcommons/power-dev/issues/220#issue-835336923
 DEVICE_TYPE_WT500 = 48
 
-MAX_RANGE_FOR_DEVICE = {
-    8: 20,
-    35: 50,
-    48: 50,
-    49: 20,
-    52: 20,
-    77: 20,
-}
-
 
 class MeasurementEndedTooFastError(Exception):
     pass
@@ -779,12 +770,7 @@ class Session:
             self._server._summary.phase("ranging", 0)
             self._ptd.start()
             self._ptd.cmd("SR,V,Auto")
-            ptd_device_type = self._server._config.ptd_device_type
-            if ptd_device_type in MAX_RANGE_FOR_DEVICE:
-                self._ptd.cmd(f"SR,A,{MAX_RANGE_FOR_DEVICE[ptd_device_type]}")
-            else:
-                logging.warning(f"Unknown max range type for device {ptd_device_type}")
-                self._ptd.cmd("SR,A,Auto")
+            self._ptd.cmd("SR,A,Auto")
             with common.sig:
                 time.sleep(ANALYZER_SLEEP_SECONDS)
             logging.info("Starting ranging mode")

--- a/ptd_client_server/lib/server.py
+++ b/ptd_client_server/lib/server.py
@@ -64,14 +64,12 @@ MULTICHANNEL_DEVICES = [48, 59, 61, 77]
 DEVICE_TYPE_WT500 = 48
 
 MAX_RANGE_FOR_DEVICE = {
-    8: 20,  # WT210
-    49: 20,  # WT310
-    52: 20,  # WT330
-    77: 20,  # WT330_multichannel
-    35: 40,  # WT500
-    48: 40,  # WT500_multichannel
-    47: 50,  # WT1800
-    66: 30,  # WT5000
+    8: 20,
+    35: 50,
+    48: 50,
+    49: 20,
+    52: 20,
+    77: 20,
 }
 
 

